### ISSUE TITLE
Support bundler 2 and 3 at the same time

### ISSUE THREAD

**Issue by izaera** [liferay/liferay-npm-tools#454]
As a consequence of https://issues.liferay.com/browse/IFI-851, we will probably need to support both bundler 3 and 2 projects in npm-scripts. Otherwise, we need to customize every bundler 3 project by hand which can lead to a big overhead in the maintenance of the codebase during the migration.

**Comment by izaera** [comment]
Currently, I'm configuring bundler 3 projects with some code like this:

- liferay-npm-bundler.config.js
```javascript
const defaults = require('liferay-npm-scripts/src/config/liferay-npm-bundler.config.js');

module.exports = {
	...defaults,
	exports: {
	},
	imports: {
		...defaults.imports
	}
}
```

As bundler 3 doesn't need presets any more (because the configuration file is `.js` and the user can do whatever he wants in it) I'm using a file inside `liferay-npm-scripts` as a kind of preset which contains the defaults and then I extend it as needed.

Now, my doubt is: 

1. should we do in npm-scripts for bundler 3 as we did for 2 and merge the project's config file with the defaults one internally? 
2. or should we do the merge in the project's side as shown above to make it more explicit, less magic, and more easily customizable?

2 has the advantage of making project configuration less opaque which can hopefully lead to people using the build being more aware of what happens under the hood. But, on the other hand, because the merge is being done at the project's side (vs. the npm-scripts side) it may lead to some people circumventing our defaults without us knowing it... :thinking: 

What do you think is best @wincent ?

**Comment by wincent** [comment]
I'd go with "1" for consistency with what we do with pretty much all other build/config files that go via `liferay-npm-scripts` (eg. see usages of [`getMergedConfig()`](https://github.com/liferay/liferay-npm-tools/search?q=getMergedConfig&unscoped_q=getMergedConfig)). I believe we merge these configs:

- `npmscripts.config.js`
- `.prettierrc.js`
- `.babelrc.js`
- `jest.config.js` (technically, we don't have any of these files in liferay-portal, but we'd merge them if we did; for some reason, we've gone with sticking the Jest config in the "jest" property of the package.json — but it gets merged just the same).
- `.eslintrc.js`
- `.npmbundlerrc` (ie. the v2 bundler)
- `stylelint.config.js` (no examples of this in liferay-portal, but if they were there, the existing code would merge them)

Basically, I think that if you make Bundler break from that pattern, it is more likely to produce confusion than deliver a benefit. I like the idea of things being less magical and more explicit, but that ship has already sailed — we pretty much bet on the opposite approach everywhere here (hiding information in an effort to increase control and make divergence from the "standard pattern" less likely).
